src/DOM.js

```diff
@@ -0,0 +1,1998 @@
+  run() {
+    if (this.isRunnable()) {
+      let running = false;
+      const srcAttr = this.attributes.src;
+      if (srcAttr) {
+        this._emit('attribute', 'src', srcAttr.value);
+        running = true;
+      }
+      if (this.childNodes.length > 0) {
+        this.innerHTML = this.childNodes[0].value;
+        running = true;
+      }
+      return running;
+    } else {
+      return false;
+    }
+  }
+}
+module.exports.HTMLScriptElement = HTMLScriptElement;
+
+class HTMLSrcableElement extends HTMLLoadableElement {
+  constructor(tagName = null, attrs = [], value = '', location = null) {
+    super(tagName, attrs, value, location);
+  }
+
+  get src() {
+    return this.getAttribute('src');
+  }
+  set src(value) {
+    this.setAttribute('src', value);
+  }
+
+  run() {
+    const srcAttr = this.attributes.src;
+    if (srcAttr) {
+      this._emit('attribute', 'src', srcAttr.value);
+      return true;
+    } else {
+      return false;
+    }
+  }
+}
+module.exports.HTMLSrcableElement = HTMLSrcableElement;
+
+class HTMLMediaElement extends HTMLSrcableElement {
+  constructor(tagName = null, attrs = [], value = '', location = null) {
+    super(tagName, attrs, value, location);
+
+    this._startTime = 0;
+    this._startTimestamp = null;
+  }
+
+  play() {
+    this._startTimestamp = Date.now();
+  }
+  pause() {}
+  load() {}
+
+  get paused() {
+    return true;
+  }
+  set paused(paused) {
+    this._startTime = this.currentTime;
+    this._startTimestamp = null;
+  }
+  get currentTime() {
+    return this._startTime + (this._startTimestamp !== null ? (Date.now() - this._startTimestamp) : 0);
+  }
+  set currentTime(currentTime) {}
+  get duration() {
+    return 1;
+  }
+  set duration(duration) {}
+  get loop() {
+    return false;
+  }
+  set loop(loop) {}
+  get autoplay() {
+    return false;
+  }
+  set autoplay(autoplay) {}
+
+  canPlayType(type) {
+    return ''; // XXX
+  }
+
+  get HAVE_NOTHING() {
+    return HTMLMediaElement.HAVE_NOTHING;
+  }
+  set HAVE_NOTHING(v) {}
+  get HAVE_METADATA() {
+    return HTMLMediaElement.HAVE_METADATA;
+  }
+  set HAVE_METADATA(v) {}
+  get HAVE_CURRENT_DATA() {
+    return HTMLMediaElement.HAVE_CURRENT_DATA;
+  }
+  set HAVE_CURRENT_DATA(v) {}
+  get HAVE_FUTURE_DATA() {
+    return HTMLMediaElement.HAVE_FUTURE_DATA;
+  }
+  set HAVE_FUTURE_DATA(v) {}
+  get HAVE_ENOUGH_DATA() {
+    return HTMLMediaElement.HAVE_ENOUGH_DATA;
+  }
+  set HAVE_ENOUGH_DATA(v) {}
+}
+HTMLMediaElement.HAVE_NOTHING = 0;
+HTMLMediaElement.HAVE_METADATA = 1;
+HTMLMediaElement.HAVE_CURRENT_DATA = 2;
+HTMLMediaElement.HAVE_FUTURE_DATA = 3;
+HTMLMediaElement.HAVE_ENOUGH_DATA = 4;
+module.exports.HTMLMediaElement = HTMLMediaElement;
+
+class HTMLSourceElement extends HTMLSrcableElement {
+  constructor(attrs = [], value = '', location = null) {
+    super('SOURCE', attrs, value, location);
+  }
+}
+module.exports.HTMLSourceElement = HTMLSourceElement;
+
+class HTMLImageElement extends HTMLSrcableElement {
```

Comment:
Posterity: these stubs are probably pointless.

The original idea was to have available a pure-JS impl of DOM with no native bindings, but... afaik nobody's ever used that feature, and these stubs aren't maintained.